storage/url/url.go

```diff
@@ -157,13 +161,15 @@ func (u *URL) Dir() string {
 
 // Join joins string and returns new URL.
 func (u *URL) Join(s string) *URL {
-	joinfn := filepath.Join
-	if u.IsRemote() {
-		joinfn = path.Join
+	joinfn := path.Join
+
+	if runtime.GOOS == "windows" {
+		s = filepath.ToSlash(s)
 	}
 
 	clone := u.Clone()
 	clone.Path = joinfn(clone.Path, s)
```

Comment:
If we don't need `filepath.Join`anymore, we may directly use `path.Join(clone.Path, s)` here.